sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/FileManager.java

```diff
@@ -232,16 +232,30 @@ public static boolean delete(@NonNull File file) {
      * @param file The file or directory to delete.
      * @return {@code true} if it was deleted, {@code false} otherwise.
      */
-    public static boolean deleteDir(File file) {
+    public static boolean deleteDirectory(@NonNull File file) {
         File[] contents = file.listFiles();
         if (contents != null) {
             for (File f : contents) {
-                deleteDir(f);
+                deleteDirectory(f);
             }
         }
         return file.delete();
     }
 
+    /**
+     * Delete all files inside this directory, without deleting directory itself.
+     *
+     * @param directory The directory to delete.
+     */
+    public static void cleanDirectory(@NonNull File directory) {
+        File[] contents = directory.listFiles();
```

Comment:
listFiles will just return null on a regular file?

Reply:
Yes, exactly. 
We also have a test, to check that is we put a regular file here, then no crashes will happen (`cleanDirectoryWhenNotDirectory`).